### ISSUE TITLE
Arguments aren't allowed after ParamSpec.args

### ISSUE THREAD

**Issue by brandonchinn178** [python/mypy#18843]
**Bug Report**

It's valid python to put arguments after `*args`, which will be keyword-only. `def foo(*args: Any, x: int)` works, but `def foo(*args: P.args, x: int)` does not.

**To Reproduce**

```python
from typing import *

def foo[T, **P](
    f: Callable[P, T],
    *args: P.args,
    x: int,
    **kwargs: P.kwargs,
) -> None:
    print(args, x, kwargs)

foo(lambda *args, **kwargs: None, 1, 2, x=3, y=4)
```

https://mypy-play.net/?mypy=latest&python=3.12&gist=22c8301eb5dbc2b6b4ba4e656b5eb48a

**Expected Behavior**

I expect this to not error

**Actual Behavior**

```
error: Arguments not allowed after ParamSpec.args  [valid-type]
```

**Your Environment**

- Mypy version used:
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used:

**Comment by brianschubert** [comment]
Hi! While that construction does not raise an error at runtime, it is disallowed by the [typing spec](https://typing.python.org/en/latest/spec/generics.html#id5). Quoting the relevant excerpt:

> * A function declared as `def inner(a: A, b: B, *args: P.args, **kwargs: P.kwargs) -> R` has type `Callable[Concatenate[A, B, P], R]`. **Placing keyword-only parameters between the `*args` and `**kwargs` is forbidden.**
> ```python
>def add(f: Callable[P, int]) -> Callable[Concatenate[str, P], None]:
>
>    def foo(s: str, *args: P.args, **kwargs: P.kwargs) -> None:  # Accepted
>        pass
>
>    def bar(*args: P.args, s: str, **kwargs: P.kwargs) -> None:  # Rejected
>        pass
>
>    return foo                                                   # Accepted
> ```

See section on [Concatenating Keyword Parameters](https://peps.python.org/pep-0612/#concatenating-keyword-parameters) from PEP 612 for more on why this construction isn't allowed.

Since mypy is acting in accordance with the typing spec here, I'm closing this as not-a-bug.

**Comment by brandonchinn178** [comment]
Thanks for the quick response! Sounds good, I guess I'll just `# type: ignore` this then...
